test/unittests/state_transition.hpp

```diff
@@ -79,7 +79,7 @@ class state_transition : public ExportableFixture
     Transaction tx{
         .gas_limit = block.gas_limit,
         .max_gas_price = block.base_fee + 1,
-        .max_priority_gas_price = 1,
+        .max_priority_gas_price = block.base_fee + 1,
```

Comment:
Why does this change? This transaction field in the spec is called `max_priority_fee_per_gas` and is what is added on top of block base fee.

Reply:
Ah I guess because default is legacy transaction? It's confusing that `Transaction` has this field for legacy.

Reply:
Or another idea is that `Transaction` constructor for legacy should enforce this

Reply:
Yes they can, but there is no time to do this change now.

Reply:
But it also can be just
```
if (tx.type == Legacy)
  tx.max_priority_gas_price = tx.max_gas_price;
```

Reply:
I improved the asserts in `TearDown()` a bit. 

It looks to me that we should have something like `TestBlock` with every field `std::optional` and set default values only after test definition. Otherwise, the test definitions must fight with defaults.